Storybook: Add BlockPatternsList story

ntsekouras opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This PR adds a simple story for `BlockPatternsList` component.

## Testing Instructions
`npm run storybook:build`

ntsekouras committed:
Storybook: Add BlockPatternsList story

ntsekouras committed:
update story with docs

ntsekouras committed:
update fixtures and pass block library styles to the block previews

youknowriad commented:
I see you're showing all the props of the component in the storybook config panel, do we really need that? 

Should we just show the actual interesting props (props to show/hide things...)

youknowriad commented:
I also wonder if we should limit the width of the story to render it like a panel (which is how we use it most times). I know we use it as a "grid" too but I prefer this to be a prop rather than the CSS hack we used today (so for later).

ciampo commented on packages/block-editor/src/components/block-patterns-list/stories/index.story.js:
```diff
@@ -0,0 +1,81 @@
+		},
+		orientation: {
+			description:
+				'Orientation for arrow key navigation. Internally Composite component is used and it"s default value.',
```
Not sure about this specific component, but usually we prefer not exposing how a component internally works, so that in the future we have more freedom on changing that aspect. Maybe you could change it to something more generic, like

```suggestion
				'Orientation for the underlying composite widget.',
```

ciampo commented on packages/block-editor/src/components/block-patterns-list/stories/index.story.js:
```diff
@@ -0,0 +1,81 @@
+	label: 'Block patterns story',
+	showTitle: true,
+	showTitlesAsTooltip: false,
+};
```
Optional, but you may want to use the most recent CSF 3: https://storybook.js.org/blog/storybook-csf3-is-here/

ciampo reviewed (commented):
A few observations:

- not all fonts in the block pattern previews seem to be loaded (especially the smaller text);
- the block previews don't load when visiting the docs page for the first time; they load when visiting the "default" example, and then they load correctly when visiting again the "docs" page;
- focus styles don't look correct unless WordPress global styles are loaded too




https://github.com/user-attachments/assets/54e82b27-0003-44ec-a7d8-395f90378181

ntsekouras committed:
Update packages/block-editor/src/components/block-patterns-list/stories/index.story.js

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

ntsekouras committed:
update story to csf3

ntsekouras commented:
>Is this the "common.css" from the block-editor package, it seems to be the same issue that I had for the zoom-out style. Another reason to bundle these styles automatically in the iframe.

It comes from core `wp-admin/common.css`.

>Personally, I think we should only show the relevant props and not all the random props, this is not a typical UI component, so the other props might not be useful in the context of storybook.

I can remove some for now and add them if necessary.

ntsekouras committed:
remove some props

ciampo commented:
> I also wonder if we should limit the width of the story to render it like a panel (which is how we use it most times). I know we use it as a "grid" too but I prefer this to be a prop rather than the CSS hack we used today (so for later).

@youknowriad we usually recommend against it, so that the component is tested "as-is". To serve this purpose, Storybook already comes with some built-in tooling to resize the viewport

![Screenshot 2024-10-21 at 17 07 42](https://github.com/user-attachments/assets/1b1c1360-6033-4c7a-b49a-4a3e577073cd)

And we have also added a custom utility to resize the component to the sidebar width, in case:

![Screenshot 2024-10-21 at 17 08 36](https://github.com/user-attachments/assets/d2738e43-e937-4d3e-90de-af0e4cc94470)

ciampo commented:
@ntsekouras 

> > focus styles don't look correct unless WordPress global styles to are loaded too
> 
> It seems in common.css we zero out the `outline` for `divs`. Do you think we should add it to the component as well? Because now the default user agent style for focus-visible adds an outline.

If it's important for the component to be "self-sufficient" in isolation, yes. Otherwise, if you're confident that the component will (for the foreseeable future) always be rendered in an environment where those global styles are loaded, then I guess we can leave as-is.

> > not all fonts in the block pattern previews seem to be loaded (especially the smaller text);
> 
> My untrained eye doesn't see a difference.. How did you get that some fonts are not loading?

I re-uploaded the video in my [original comment](https://github.com/WordPress/gutenberg/pull/66227#pullrequestreview-2381862996) — I refer to this part:

![Screenshot 2024-10-21 at 17 11 37](https://github.com/user-attachments/assets/8c922bd4-832d-4d10-b373-c61ae4503e5a)

ntsekouras committed:
add outline and bring back props

ntsekouras commented:
I added the outline and brought back the props I removed in the previous commit.

Regarding the loading if we refresh I think it's related to storybook - maybe something with lazy loading? In the video below you can see that the blockPreview file is not loaded and if I just hover at the storybook sidebar it does load..

https://github.com/user-attachments/assets/44a29b73-dd8c-4098-89b9-9ade61cb2b30

youknowriad commented:
Can you clarify why we're keeping all these props, for me they create more confusion than solve anything for the user. 

 - Do they bring any value to the user visiting storybook?
 - Do they bring any value to the developer iterating on the component in storybook?

ntsekouras commented:
>Can you clarify why we're keeping all these props, for me they create more confusion than solve anything for the user.

>Do they bring any value to the user visiting storybook?
> Do they bring any value to the developer iterating on the component in storybook?

For now with just a single simple story, probably no. But if we add more to them, it should be helpful.

Why do you think it's so unnecessary or confusing though?

youknowriad commented:
> Why do you think it's so unnecessary or confusing though?

I like to curate what shows up in storybook. This component is an internal component, it's prefixed, so it was never meant to be public, which means we never really thought about these props as public props for users to use. I'd rather be explicit there and only show what we know is a useful prop. Also not sure what event props and things like that bring to a story if they can't be used.

youknowriad commented:
We could also keep the props that actually have an impact (like labels, showing titles, tooltips, draggable...),  the ones we know we want to keep.

youknowriad reviewed (approved):
I realized that I'm being too finicky for a PR that adds a story.

Let's ship :)

ntsekouras commented:
>I realized that I'm being too finicky for a PR that adds a story.

😄 

Let's curate with the iterations on this component to be reused in a more consistent way.
